file_organizer_v2/src/file_organizer/models/audio_transcriber.py

```
@@ -0,0 +1,417 @@
+            )
+
+            return LanguageDetection(
+                language=language, language_name=language_name, confidence=confidence
+            )
+
+        except Exception as e:
+            logger.error(f"Language detection failed: {e}")
+            raise RuntimeError(f"Language detection failed: {e}") from e
+
+    def transcribe(
+        self, audio_path: Union[str, Path], options: Optional[TranscriptionOptions] = None
```

Ruff (0.14.13)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)